lib/Conversion/BGVToOpenfhe/BGVToOpenfhe.cpp

```diff
@@ -0,0 +1,178 @@
+#include "include/Conversion/BGVToOpenfhe/BGVToOpenfhe.h"
+
+#include "include/Dialect/BGV/IR/BGVDialect.h"
+#include "include/Dialect/BGV/IR/BGVOps.h"
+#include "include/Dialect/BGV/IR/BGVTypes.h"
+#include "include/Dialect/LWE/IR/LWEAttributes.h"
+#include "include/Dialect/LWE/IR/LWEDialect.h"
+#include "include/Dialect/LWE/IR/LWEOps.h"
+#include "include/Dialect/LWE/IR/LWETypes.h"
+#include "include/Dialect/Openfhe/IR/OpenfheDialect.h"
+#include "include/Dialect/Openfhe/IR/OpenfheOps.h"
+#include "include/Dialect/Openfhe/IR/OpenfheTypes.h"
+#include "lib/Conversion/Utils.h"
+#include "mlir/include/mlir/Dialect/Func/IR/FuncOps.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/DialectConversion.h"  // from @llvm-project
+
+namespace mlir::heir::bgv {
+
+#define GEN_PASS_DEF_BGVTOOPENFHE
+#include "include/Conversion/BGVToOpenfhe/BGVToOpenfhe.h.inc"
+
+class ToLWECiphertextTypeConverter : public TypeConverter {
+ public:
+  // Convert ciphertext to RLWE ciphertext
+  ToLWECiphertextTypeConverter(MLIRContext *ctx) {
+    addConversion([](Type type) { return type; });
+    addConversion([ctx](CiphertextType type) -> Type {
+      assert(type.getLevel().has_value());
+      auto level = type.getLevel().value();
+      assert(level < type.getRings().getRings().size());
+
+      auto ring = type.getRings().getRings()[level];
+      auto cmod = ring.getCmod();
+      auto polyDegree = ring.getIdeal().getDegree();
+      auto dim = type.getDim();
+      return lwe::RLWECiphertextType::get(
+          // FIXME: What should be the default encoding?
```

Comment:
I'm guessing this TODO needs to be resolved first to enable this: https://github.com/google/heir/blob/e83489c16703eb969757c59cd48785b388097bea/include/Dialect/BGV/IR/BGVTypes.td#L37

https://github.com/google/heir/issues/99

Given the parameter corresponding to the set of rings, I'm not so sure anymore about the comment

> I think we could perhaps also migrate the bgv::Ciphertext type to the lwe dialect as an RLWE ciphertext type

But if the BGVCiphertext type stored the number of bits of the underlying message, then I believe that would be sufficient information to choose a default encoding via `#lwe.polynomial_evaluation_encoding<cleartext_start=K, cleartext_bitwidth=B>` for some `K` and `B` indexing the coefficient space Z/qZ

Two notes:

- I know that BGV cane use either RLWE_PolynomialCoefficientEncoding or RLWE_PolynomialEvaluationEncoding, but IIUC RLWE_PolynomialEvaluationEncoding is the most commonly used one, so that should be the default.
- There is definitely a lot more discussion to be had about how to pick an encoding, and I think the simplest thing to do for now is to pick any sensible default we can, file an issue for further discussion, and (optional, perhaps in followup PR) allow the user to manually override the default choice as a pass option.